RecoTracker/LST/plugins/LSTOutputConverter.cc

```diff
@@ -0,0 +1,153 @@
+  //desc.addUntracked<edm::InputTag>("beamSpot", edm::InputTag("offlineBeamSpot"));
+
+  //edm::ParameterSetDescription psd0;
+  //psd0.add<std::string>("ComponentName", std::string("SeedFromConsecutiveHitsCreator"));
+  //psd0.add<std::string>("propagator", std::string("PropagatorWithMaterial"));
+  //psd0.add<double>("SeedMomentumForBOFF", 5.0);
+  //psd0.add<double>("OriginTransverseErrorMultiplier", 1.0);
+  //psd0.add<double>("MinOneOverPtError", 1.0);
+  //psd0.add<std::string>("magneticField", std::string(""));
+  //psd0.add<std::string>("TTRHBuilder", std::string("WithTrackAngle"));
+  //psd0.add<bool>("forceKinematicWithRegionDirection", false);
+  //desc.add<edm::ParameterSetDescription>("SeedCreatorPSet", psd0);
+
+  descriptions.addWithDefaultLabel(desc);
+}
+
+void LSTOutputConverter::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  // Setup
+  auto const& lstOutput = iEvent.get(lstOutputToken_);
+  auto const& phase2OTRecHits = iEvent.get(lstPhase2OTHitsInputToken_);
+  auto const& pixelSeeds = iEvent.get(lstPixelSeedToken_);
+  const auto& mf = iSetup.getData(mfToken_);
+  //auto const& bs = iEvent.get(beamSpotToken_);
+
+  // Vector definitions
+  std::vector<std::vector<unsigned int>> const lstTC_hitIdx = lstOutput.hitIdx();
+  std::vector<unsigned int> const lstTC_len = lstOutput.len();
+  std::vector<float> const lstTC_pt = lstOutput.pt();
+  std::vector<float> const lstTC_eta = lstOutput.eta();
+  std::vector<float> const lstTC_phi = lstOutput.phi();
+  std::vector<int> const lstTC_seedIdx = lstOutput.seedIdx();
+
+  TrackCandidateCollection output;
+  output.reserve(lstTC_len.size());
+
+  auto const& OTHits = phase2OTRecHits.hits();
+
+  //auto seedsFromT5 = std::make_unique<TrajectorySeedCollection>();
+  //GlobalPoint vtxOfBs(bs.x0(), bs.y0(), bs.z0()); 
+  for (unsigned int i=0; i<lstTC_len.size(); i++) {
+    if (lstTC_len[i]==10) { // T5
+      continue; // temporary
+      //std::vector<Hit> hitsFromT5;
+      //for (auto hitIdx : lstTC_hitIdx[i])
+      //  hitsFromT5.push_back((Hit) OTHits[hitIdx]);
+      //sort(hitsFromT5.begin(), hitsFromT5.end(), HitLessByRadius());
+      //GlobalTrackingRegion region(lstTC_pt[i], vtxOfBs, 0.2, 0.2);
+      //seedCreator_->init(region, iSetup, nullptr);
+      //seedCreator_->makeSeed(*seedsFromT5, hitsFromT5);
+    }
+
+    //TrajectorySeed s = lstTC_len[i]==10 ? (*seedsFromT5)[0] : pixelSeeds[lstTC_seedIdx[i]];
+    TrajectorySeed s = pixelSeeds[lstTC_seedIdx[i]];
+    //PTrajectoryStateOnDet st = s.startingState();
+
+    edm::OwnVector<TrackingRecHit> recHits;
+    for (auto const& hit : s.recHits())
+      recHits.push_back(hit.clone());
+
+    TrajectoryStateOnSurface tsos = trajectoryStateTransform::transientState(s.startingState(), recHits.back().surface(), &mf);
+    PTrajectoryStateOnDet st = trajectoryStateTransform::persistentState(tsos, recHits[0].det()->geographicalId().rawId());
```

Comment:
this is still inconsistent: tsos is defined on the last hit of the pLS, while the detector is from the first hit (what's saved is a local position of the tsos on the outer module and the detid saying that it's the inner module). So, very likely the state is shifted and none of the hits can be picked up